plugins/modules/source_control/git/git_commit.py

```diff
@@ -0,0 +1,243 @@
+options:
+    path:
+        description:
+            - Folder path where C(.git/) is located.
+        required: true
+        type: path
+    comment:
+        description:
+            - Git commit comment. Same as C(git commit -m).
+        type: str
+        required: true
+    add:
+        description:
+            - List of files under C(path) to be staged. Same as C(git add .).
+              File globs not accepted, such as C(./*) or C(*).
+        type: list
+        elements: str
+    empty_commit:
+        description:
+            - Drive module behaviour in case nothing to commit.
+              If C(allow) empty commit is allowed, same as C(--allow-empty).
+              Do not commit if C(skip). Fail job if C(fail).
+        choices: ['allow', 'fail', 'skip']
+        type: str
+        default: 'fail'
+requirements:
+    - git>=2.10.0 (the command line tool)
+'''
+
+EXAMPLES = '''
+- name: Add and commit two files.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: My amazing backup
+    add: ['test.txt', 'txt.test']
+    empty_commit: fail
+
+- name: Empty commit.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: My amazing empty commit
+    empty_commit: allow
+    add: ['.']
+
+- name: Skip if nothing to commit.
+  community.general.git_commit:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: Skip my amazing empty commit
+    empty_commit: skyp
+    add: ['.']
+'''
+
+RETURN = '''
+output:
+    description: list of git cli command stdout
+    type: list
+    returned: always
+    sample: [
+        "[master 99830f4] Remove [ test.txt, tax.txt ]\n 4 files changed, 26 insertions(+)..."
+    ]
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.utils.display import Display
+
+display = Display()
+
+
+def git_add(module):
+
+    add = module.params.get('add')
+    path = module.params.get('path')
+
+    if not add:
+        add = ["."]
+    # "git add do_not_exist" -> rc: 128
+    # "git add i_exist"      -> rc: 0
+    # "git add ."            -> rc: 0
+    add_cmds = [
+        'git',
+        'add',
+    ]
+    for item in add:
+        add_cmds.insert(len(add_cmds), item)
+
+    rc, _output, error = module.run_command(add_cmds, cwd=path)
+
+    if rc == 128:
+        module.fail_json(msg=error)
+
+    if rc == 0:
+        # no output returned when rc = 0
+        return
+
+
+def git_commit(module):
+
+    def git_commit_run(cmd, path):
+
+        rc, output, error = module.run_command(cmd, cwd=path)
+
+        if rc == 1:
+            module.fail_json(msg=error)
+        if rc == 0:
+            return output
+
+    result = dict(changed=False)
+    empty_commit = module.params.get('empty_commit')
+    comment = module.params.get('comment')
+    path = module.params.get('path')
+    cmd_porcelain = ['git', 'commit', '--porcelain']
```

Comment:
This command does not work, the same as `git commit` it self will not work either. You need more command line arguments.
```suggestion
    cmd_porcelain = ['git', 'commit', '--porcelain']
```

Reply:
Why not? Is it not `--porcelain` purpose to see if there are files to be `add` and `commit`? 
```
(ansible_collections) olivierif:NaC federicoolivieri$ git commit --porcelain
(ansible_collections) olivierif:NaC federicoolivieri$ touch tests.txt
(ansible_collections) olivierif:NaC federicoolivieri$ git commit --porcelain
?? tests.txt
(ansible_collections) olivierif:NaC federicoolivieri$ echo $?
1
```